tasks/generic_validation.yml

```diff
@@ -2,3 +2,8 @@
   fail:
     msg: "Customer name length can not be higher than 6 characters. Current length is {{ customer | length }}."
   when: customer | length > 6
+
+- name: check if env_type is a valid environment
+  fail:
+    msg: "defined env_type {{ env_type }} is not a valid environment"
+  when: environments | json_query("[?name=='"~ env_type ~"'] | [0]") is none
```

Comment:
I would argue that "generic" would be a good name if it was playbook that can be run by several callers with different inputs. In this case it looks like a validation playbook with all the specific checks inside for everybody right? If yes, maybe just call it "validate_configuration"? "validate_inputs"? The use of an infinitive is in line with other playbooks: deploy, switch, etc.

Reply:
I vote for `validate_configuration`

Reply:
for now we only have configuration but this is not the purpose of this file. Its purpose is to contain generic pre_task to not have to create a new one and re-include it in all existing playbooks.

Reply:
My point was that the word "generic" is usually used for a shared abstraction. Here it's just a task that you play in many playbooks to "validate configuration".

Reply:
there is no notion of abstraction or interface or other word with ansible.
The file is renamed.